src/main/java/individual/freshplace/config/auth/PrincipalDetailsService.java

```diff
@@ -0,0 +1,23 @@
+package individual.freshplace.config.auth;
+
+import individual.freshplace.entity.Member;
+import individual.freshplace.repository.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class PrincipalDetailsService implements UserDetailsService {
+
+    private final MemberRepository memberRepository;
+    private static UsernameNotFoundException exception = new UsernameNotFoundException("사용자를 찾을 수 없습니다.");
+
+    @Override
+    public UserDetails loadUserByUsername(String memberId) throws UsernameNotFoundException {
+        Member memberEntity = memberRepository.findByMemberId(memberId).orElseThrow(() -> exception);
```

Comment:
```suggestion
        Member memberEntity = memberRepository.findByMemberId(memberId)
            .orElseThrow(() -> new UsernameNotFoundException("사용자를 찾을 수 없습니다."));
```
제생각에는 굳이 exception을 상수로 뺄 필요는 없어보여요.
그렇다고 상수의 네이밍이 exception으로 큰 의미를 가지고 있지도 않구요. 어떻게 생각하세요?

Reply:
> 제생각에는 굳이 exception을 상수로 뺄 필요는 없어보여요. 그렇다고 상수의 네이밍이 exception으로 큰 의미를 가지고 있지도 않구요. 어떻게 생각하세요?

일단 변수명은 의미를 가지도록 usernameNotFoundException로 변경하는게 좋을것같습니다. !!